platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/StateSigner.java

```diff
@@ -64,10 +64,11 @@ public StateSigner(@NonNull final HashSigner signer, @NonNull final PlatformStat
 
             final Hash stateHash =
                     Objects.requireNonNull(reservedSignedState.get().getState().getHash());
-            final Signature signature = signer.sign(stateHash);
+            final Bytes signature = signer.signImmutable(stateHash);
             Objects.requireNonNull(signature);
 
-            return new StateSignatureTransaction(reservedSignedState.get().getRound(), signature, stateHash);
+            return new StateSignatureTransaction(
+                    reservedSignedState.get().getRound(), signature, stateHash.getBytes(), Bytes.EMPTY);
```

Comment:
Presumably the end goal is not to have a `Bytes.EMPTY` epoch hash. Could you leave a future work comment explaining what still needs to be done here?